### PR TITLE
fix(avatars): prevent flicker when loading avatars on settings pages

### LINKED ISSUE #2105: prevent flickr when showing avatar in settings
We should be smarter about displaying the profile image. We can store the image URL in localStorage as a hint for what to initially display while we determine the latest state of the account's profile image.

### PR DESCRIPTION
**Pull request by zaach**
There's currently a brief flash of the default profile image before your custom profile image is shown. This PR hides the default image unless it's needed.

Fixes #2105.

@shane-tomlinson or @vladikoff r?

### CODE REVIEW

File: app/scripts/views/mixins/settings-mixin.js [hunk 1]
```diff
@@ -33,6 +33,17 @@ define([
         Session.clear();
         self.user.clearSignedInAccount();
       }
+    },
+
+    // Cache the profile image URL whenever it changes
+    // TODO: issue #2095 send broadcast
+    onAvatarChange: function (avatar) {
```
Comment (shane-tomlinson) [inline app/scripts/views/mixins/settings-mixin.js]: Two comments:
* It seems like this functionality belongs in the avatar-mixin. 
* `onAvatarChange` hints that this function is called in response to an avatar change, on a model somewhere. The functionality is only kinda-sorta that, and also sets an avatar on the account and user. Maybe rename the function `updateAvatarUrl` and have it take just a URL?

**Comment by shane-tomlinson** [comment]
@zaach - I'm pretty sure this is user error somehow, whenever I set the avatar from the local camera, the photo appears to upload, but then is not found when requested from the profile server.

screen shot 2015-02-18 at 12 18 22

Looking at the profile server's logs, I see:

screen shot 2015-02-18 at 12 21 20

The profile server's worker logs are:

screen shot 2015-02-18 at 12 22 34

Any hints?

**Comment by shane-tomlinson** [comment]
An issue I came across is the profile image URL is still set in the account model after the user deletes the image, so the default image is only shown after the request to the profile server completes.

### STR

1. Add a profile image
1. Delete the profile image
1. set a breakpoint https://github.com/mozilla/fxa-content-server/pull/2106/files#diff-69c1b4c151f87a5c9ad048f78656cf9aR48
1. reload the settings page, the contents of the if loop are skipped.
1. reload the settings page, go to the dev tools console and do an `account.get('profileImageUrl')` - the gravatar URL is returned.

screen shot 2015-02-18 at 12 47 03

**Comment by shane-tomlinson** [comment]
@zaach - Can you take a screen cap of the effect being eliminated and add it to the original bug for posterity? I am able to test/see the delay if I manually add a delay to the avatar endpoint response on the profile server.

**Comment by zaach** [comment]
@shane-tomlinson looks like a profile server issue but I'm not too familiar with the implementation.

**Comment by zaach** [comment]
How it looks with the fix: blah

**Comment by shane-tomlinson** [comment]
Filed follow on issues #2116 and #2117

**Pull request merged**